scrapy/core/downloader/handlers/http2.py

```diff
@@ -0,0 +1,142 @@
+        self._crawler = crawler
+
+        from twisted.internet import reactor
+        self._pool = H2ConnectionPool(reactor, settings)
+        self._context_factory = load_context_factory_from_settings(settings, crawler)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler.settings, crawler)
+
+    def download_request(self, request: Request, spider: Spider):
+        agent = ScrapyH2Agent(
+            context_factory=self._context_factory,
+            pool=self._pool,
+            crawler=self._crawler
+        )
+        return agent.download_request(request, spider)
+
+    def close(self) -> None:
+        self._pool.close_connections()
+
+
+class ScrapyProxyH2Agent(H2Agent):
+    def __init__(
+        self, reactor: ReactorBase,
+        proxy_uri: URI, pool: H2ConnectionPool,
+        connect_timeout: Optional[float] = None, bind_address: Optional[bytes] = None
+    ) -> None:
+        super(ScrapyProxyH2Agent, self).__init__(
+            reactor=reactor,
+            pool=pool,
+            connect_timeout=connect_timeout,
+            bind_address=bind_address
+        )
+        self._proxy_uri = proxy_uri
+
+    @staticmethod
+    def get_key(uri: URI) -> Tuple:
+        return "http-proxy", uri.host, uri.port
+
+
+class ScrapyH2Agent:
+    _Agent = H2Agent
+    _ProxyAgent = ScrapyProxyH2Agent
+
+    def __init__(
+        self, context_factory,
+        connect_timeout=10,
+        bind_address: Optional[bytes] = None, pool: H2ConnectionPool = None,
+        crawler=None
+    ) -> None:
+        self._context_factory = context_factory
+        self._connect_timeout = connect_timeout
+        self._bind_address = bind_address
+        self._pool = pool
+        self._crawler = crawler
+
+    def _get_agent(self, request: Request, timeout: Optional[float]) -> H2Agent:
+        from twisted.internet import reactor
+        bind_address = request.meta.get('bindaddress') or self._bind_address
+        proxy = request.meta.get('proxy')
+        if proxy:
+            _, _, proxy_host, proxy_port, proxy_params = _parse(proxy)
+            scheme = _parse(request.url)[0]
+            proxy_host = str(proxy_host, 'utf-8')
+            omit_connect_timeout = b'noconnect' in proxy_params
+            if omit_connect_timeout:
+                warnings.warn("Using HTTPS proxies in the noconnect mode is deprecated. "
+                              "If you use Crawlera, it doesn't require this mode anymore, "
+                              "so you should update scrapy-crawlera to 1.3.0+ "
+                              "and remove '?noconnect' from the Crawlera URL.",
+                              ScrapyDeprecationWarning)
+
+            if scheme == b'https' and not omit_connect_timeout:
+                proxy_auth = request.headers.get(b'Proxy-Authorization', None)
+                proxy_conf = (proxy_host, proxy_port, proxy_auth)
+
+                # TODO: Return TunnelingAgent instance
```

Comment:
I think we should raise `NotImplementedError`, and cover this in the relevant parts of the documentation, explaining that the new downloader handler does not support this at the moment.

It should also make the `else:` below unnecessary, and allow for dedenting its `return` statement.